src/Interpreters/Context.cpp

```diff
@@ -1057,14 +1062,14 @@ ConfigurationPtr Context::getUsersConfig()
     return shared->users_config;
 }
 
-void Context::setUser(const UUID & user_id_)
+void Context::setUser(const UUID & user_id_, bool initialize_quota_)
```

Comment:
Here again, it is not clear what is going on with `initialize_quota`. Maybe add some comments to the source code.

Reply:
I added more comments to code. 
[Here ](https://github.com/ClickHouse/ClickHouse/pull/51724#discussion_r1266011339) You can read about the reason why intialize_quota_ was added